### PR TITLE
8599 images can be added in summernote

### PR DESCRIPTION
**Pull request by arisah-ijeoma**

### CODE REVIEW

**Comment by emitche** [comment]
The failed spec is unrelated to the changes.

**Comment by arisah-ijeoma** [comment]
the pr depends on #548 8598

it has to be merged into it first

I have already set it as the base branch for this pr

**Comment by emitche** [comment]
Thanks for the update!

**Comment by emitche** [comment]
The styling is not saving on the images, just the images themselves. You might need to add the attributes `style src data-filename`.

File: app/views/admin/loans/questionnaires/_questionnaire_group.html.slim [hunk 1]
```diff
@@ -84,12 +84,14 @@
               = f.input_field :"#{question.attribute_sym}[number]", as: :decimal,
                 value: response.number
 
-            - tags = %w(table tbody tr td b i u ol ul li p)
+            - tags = %w(table tbody tr td b i u ol ul p img)
```
Comment (emitche) [inline app/views/admin/loans/questionnaires/_questionnaire_group.html.slim]: We need to add `li` here.

**Comment by emitche** [comment]
I didn't do a QA review on this yet. But there is a code revision that is needed because of changes since the last merge.

File: app/views/admin/loans/questionnaires/_questionnaire_group.html.slim (hunk 1 above)
Comment (arisah-ijeoma) [reply app/views/admin/loans/questionnaires/_questionnaire_group.html.slim]: the lists were not saving appropriately

unordered and ordered @emitche

Comment (emitche) [reply app/views/admin/loans/questionnaires/_questionnaire_group.html.slim]: Oh. So we need to not include `li`?

Comment (emitche) [reply app/views/admin/loans/questionnaires/_questionnaire_group.html.slim]: I've tested the current functionality on staging. v2.11.1 uses this line: `- tags = %w(table tbody tr td b i u ol ul li p)` So we do need to include `li` in this list.

I will make that change.

**Comment by emitche** [comment]
Saved tables do not display properly.

#### Edit view
image

#### Show view
image

This is how the table should display:

#### Edit view
(The last cell was initially D3 and changed to D34 in later images.)
image

#### Show view
image

**Comment by emitche** [comment]
I am still working on QA for this ticket. It will be moved to a different sprint.

**Comment by emitche** [comment]
summernote-init.js probably should be this when resolving the merge conflicts:

```
        ["style", ["bold", "italic", "underline", "clear"]],
        ["table", ["table"]],
        ["para", ["ul", "ol", "paragraph"]],
        ["insert", ["link", "picture"]]
```

**Comment by emitche** [comment]
QA review:

The image is not saving properly when tested in the UI. Instead, in the html, an empty `` tag exists.

I think the problem is that attributes are not enabled in the santitizer. These attributes are required to properly display the image. At a minimum, we need to include the attributes: style, src, and data-filename. We may also want to include _moz_resizing.

What shows in the html when I add an image into a loan memo response for Summernote while in the edit form view before saving:
image

What shows in the html when I save the form: (No image displays in the UI.)
image

**Comment by emitche** [comment]
Test failure is a known failure unrelated to this ticket.

**Comment by emitche** [comment]
I'd like to hold on merging this work until the Summernote work for documentation is completed.

**Pull request merged**